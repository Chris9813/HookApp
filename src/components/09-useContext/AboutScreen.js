import React, { useContext } from 'react'
import { UserCotext } from './UserContext'


export const AboutScreen = () => {


    const {user, setuser} = useContext(UserCotext)

    const handleClick = () => {
        setuser({})
    }

    return(
        <div>
            <h1>About Screen</h1>
            <hr/>
            <pre>
                {JSON.stringify(user, null , 3)}
            </pre>
            <button
            className = "btn btn-warning"
            onClick = {handleClick}
            >
                Logout
            </button>
        </div>
    )

    /*
    const {user, setuser} = useContext(UserCotext)

    const handleClick = () => {
        setuser({})
    }

    return (
        <div>
            <h1>About Screen</h1>
            <hr/>
            <pre>
                {JSON.stringify(user,null,3)}
            </pre>
            <button
            className = "btn btn-warning"
            onClick = {handleClick}
            >
            Logout
            </button>
        </div>
    )
*/
}
