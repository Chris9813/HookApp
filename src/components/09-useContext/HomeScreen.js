import React, {useContext} from 'react'
import { UserCotext } from './UserContext'

export const HomeScreen = () => {

    const {user} = useContext(UserCotext)
    console.log(user);
    return (
        <div>
            <h1>Home Screen</h1>
            <hr/>
            <div>
                <pre>
                    {JSON.stringify(user, null, 3)}
                </pre>
            </div>
        </div>
    )
}

