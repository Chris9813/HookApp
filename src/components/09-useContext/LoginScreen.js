import React, { useContext } from 'react'
import { UserCotext } from './UserContext'

export const LoginScreen = () => {


    const {setuser} = useContext(UserCotext)

    return(
        <div>
            <h1>Login Screen</h1>
            <hr/>
            <div>
                <button
                className = "btn btn-primary"
                onClick = {() => {setuser({
                    id: 123,
                    name: "Christian",
                })}}
                >
                    Login
                </button>
            </div>
        </div>
    )

    /*
    const {setuser} = useContext(UserCotext)

    return (
        <div>
        <div>
            <h1>Login Screen</h1>
            <hr/>
            <button
            className = "btn btn-primary"
            onClick = {() => setuser({
                id: 123,
                name: "Christian"
            })}
            >
                Login
            </button>
        </div>
        </div>
    )
*/
}
