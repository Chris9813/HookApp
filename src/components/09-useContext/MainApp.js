import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserCotext } from './UserContext'

export const MainApp = () => {

    const [user, setuser] = useState({})

    return (
        <UserCotext.Provider
        value = {{
            user,
            setuser
        }}
        >
            <AppRouter />
        </UserCotext.Provider>
    )

    /*
    const [user, setuser] = useState({})
    return (
        <UserCotext.Provider value = {{
            user,
            setuser
        }}>
            <AppRouter />
        </UserCotext.Provider>
        
    )
*/
}
