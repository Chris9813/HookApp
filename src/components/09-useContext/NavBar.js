import React from 'react'
import {NavLink} from "react-router-dom"

export const NavBar = () => {
    return (
<>
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
    <div className="container-fluid">
    <a className="navbar-brand" href="#">useContext</a>
    <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
    <div className="navbar-nav">
        <NavLink exact activeClassName = "nav-link active" className="nav-link" to="/">Home</NavLink>
        <NavLink exact  activeClassName = "nav-link active" className="nav-link" to="/login">Login</NavLink>
        <NavLink exact  activeClassName = "nav-link active" className="nav-link" to="/about">About</NavLink>
    </div>
    </div>
</div>
</nav>
</>
)
}


/*
    <NavLink exact activeClassName = "list-group-item active" className="navbar-brand" to="/">Home</NavLink>
    <NavLink exact  activeClassName = "list-group-item active" className="navbar-brand" to="/login">Login</NavLink>
    <NavLink exact  activeClassName = "list-group-item active" className="navbar-brand" to="/about">About</NavLink>
*/


/*

*/