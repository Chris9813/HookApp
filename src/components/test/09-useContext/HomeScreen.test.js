import { mount, shallow } from 'enzyme'
import React from 'react'
import { HomeScreen } from '../../09-useContext/HomeScreen'
import { UserCotext } from '../../09-useContext/UserContext'


describe('pruebas en HomeSCreen', () => {

    const user = {
        name: "Chris",
        email: "sada"
    }
    const wrapper = mount(
    <UserCotext.Provider value = {{
        user
    }}>
        <HomeScreen /> 
    </UserCotext.Provider>
        
    )

    test('debe mostrarse corectamente ', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
})
