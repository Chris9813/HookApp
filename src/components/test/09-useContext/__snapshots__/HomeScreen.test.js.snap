// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en HomeSCreen debe mostrarse corectamente  1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <div>
    <pre>
      {
   "name": "Chris",
   "email": "sada"
}
    </pre>
  </div>
</div>
`;
